### ISSUE TITLE
[bugfix-2.0.x]. If TMC2130 is enabled Marlin crashes very soon even before Init ()

### ISSUE THREAD

**Issue by mr-miky** [MarlinFirmware/Marlin#12175]
It's been more than a week since I've been trying to figure out why MARLIN does not want to work when change the config file from:
```
#define E0_DRIVER_TYPE A4988
// # define E0_DRIVER_TYPE TMC2130
```
to :
```
// # define E0_DRIVER_TYPE A4988
#define E0_DRIVER_TYPE TMC2130
```
With the A4988 driver everything works fine the logo appears on the display and then from the main menu you can operate as usual.
With  TMC2130 driver does not even arrive at the Init ().
A species of exception is created that loop the STM32F4.
I have identified where it happens but it is not clear to me.

On **ultralcd_impl_DOGM.h**
```
// The current graphical page being rendered
u8g_page_t & page = ((u8g_pb_t *) ((u8g.getU8g ()) -> dev-> dev_mem)) -> p;
```
in particular, dev_mem is 0 when TMC2130 is enabled.
I bypassed the obstacle by declaring:

`u8g_page_t * page = 0;`

and then I set the value in:

**ultralcd_impl_DOGM.h**
```
// Initialize or re-initialize the LCD
// static void lcd_implementation_init () {
void lcd_implementation_init () {

if (page == 0) {
   page = & ((u8g_pb_t *) ((u8g.getU8g ()) -> dev-> dev_mem)) -> p;
}
```
I obviously modified the rest of the code to use the pointer.

With the A4988 driver everything works, with the TMC no but at least I can get to the setup ().

Attached to the debugger snapshots.
page is zero because the debugger is stopped at main () and lcd_implementation_init () has not yet been called
![bb](https://user-images.githubusercontent.com/12703876/47267705-b9b76b80-d547-11e8-8f72-e02a28584be8.jpg)

**Comment by mr-miky** [comment]
Obviously no optimization (Debug / -Og)

**Comment by thinkyhead** [comment]
Try compiling with different versions of Arduino IDE and PlatformIO to see if it makes any difference. This could be an issue with the TMCStepper library also. See if newer versions of that library are available in the library manager.

**Comment by mr-miky** [comment]
Two weeks of research with the debugger, and I found that at the end of __static_initialization_and_destruction_0 () at ultralcd_impl_DOGM.h: something was going to alter the damn pointer.

All caused by the fact that the buld occurred with TMCStepper library version 0.21 although I had selected 0.22. At one point I decided to delete the TMCStepper library 0.21 and everything works since then.

It would be interesting to be able to insert a verification in Sanitychech.h for all those libraries that practice "black magic".
